### PR TITLE
fix: update `Requeue` and `Reassign` logic to fix performance degradation when many events are queued

### PR DESCRIPTION
**Pull request by abelanger5**
# Description

Logic for requeueing and reassigning did not limit the number of step runs to requeue, so when events accumulate with no worker present it causes memory to spike along with a very high query latency on the database. This PR primarily limits the number of step runs returned in the requeue and reassign queries, and also properly locks step run rows for these queries so only a step run in a `PENDING` or `PENDING_ASSIGNMENT` state can be requeued. 

This PR also improves the logging for the load test suite as there were simply too many logs generated to figure out the source of duplicated step runs. 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Refactor (non-breaking changes to code which doesn't change any behaviour)
- [X] Test changes (add, refactor, improve or change a test)

### CODE REVIEW

File: internal/repository/prisma/dbsqlc/step_runs.sql [hunk 1]
```diff
@@ -274,105 +275,186 @@ FROM step_run_data
...
+    FROM
+        step_runs sr
+    FOR UPDATE SKIP LOCKED
+)
+UPDATE
+    "StepRun"
+SET
+    "status" = 'PENDING_ASSIGNMENT',
+    -- requeue after now plus 4 seconds
+    "requeueAfter" = CURRENT_TIMESTAMP + INTERVAL '4 seconds',
```
Comment (steebchen) [inline internal/repository/prisma/dbsqlc/step_runs.sql:355]: not really related to this PR but I'm wondering if we should make this configurable so we can adapt this in the load test to see if it has any influence on the general reliability and performance?

File: internal/repository/prisma/step_run.go [hunk 2]
```diff
@@ -658,6 +700,21 @@ func (s *stepRunEngineRepository) updateStepRunCore(
...
+			Tenantid:  sqlchelpers.UUIDFromStr(tenantId),
+		})
+
+		if err != nil {
+			// not a fatal error if there's not a semaphore to update
+			if !errors.Is(err, pgx.ErrNoRows) {
+				return nil, fmt.Errorf("could not upsert worker semaphore: %w", err)
+			}
+		}
+	}
```
Comment (steebchen) [inline internal/repository/prisma/step_run.go]:
```suggestion
		// not a fatal error if there's not a semaphore to update
		if err != nil && !errors.Is(err, pgx.ErrNoRows) {
			return nil, fmt.Errorf("could not upsert worker semaphore: %w", err)
		}
	}
```

**Review by steebchen — APPROVED** [review]

File: internal/services/controllers/jobs/controller.go [hunk 3]
```diff
@@ -556,30 +547,38 @@ func (ec *JobsControllerImpl) runStepRunReassignTenant(ctx context.Context, tena
...
-			var innerStepRun *dbsqlc.GetStepRunForEngineRow
-
 			ctx, span := telemetry.NewSpan(ctx, "handle-step-run-reassign-step-run")
 			defer span.End()
 
-			stepRunId := sqlchelpers.UUIDToStr(stepRunCp.ID)
+			stepRunId := sqlchelpers.UUIDToStr(stepRunCp.StepRun.ID)
+
+			// if the current time is after the scheduleTimeoutAt, then mark this as timed out
+			now := time.Now().UTC().UTC()
```
Comment (grutt) [inline internal/services/controllers/jobs/controller.go:556]: double UTC?

**Review by grutt — APPROVED** [review]

File: internal/repository/prisma/dbsqlc/step_runs.sql (hunk 1 above)
Comment (abelanger5) [reply internal/repository/prisma/dbsqlc/step_runs.sql:355]: Yep will file an issue

**Pull request merged**